### DISCUSSION TITLE
Single Channel Works, Multi Channel doesn't.-solved

### DISCUSSION THREAD

**Discussion by stretch911** [PlummersSoftwareLLC/NightDriverStrip#188] [General]
i am still getting driver/adc_deprecated.h:14 errors, but thanks to some reading here, I've got (sorta) my heltec v2 working w/ a few effects. I can use DEMO just fine, (sorta depending on what effects i use). I want this code to work with 4 Channels for the Atom Fire Lamp. However, every time I up the Channel count from 1 to 2 or to 4, things stop working. Its almost like GPIO5 just dies...any help?

 Not sure why either, some of the effects just blow up...i think...drawing too much power despite low brightness setting and limited power set. Example is new ColorFillEffect(CRGB::White, 1)...if i run ATOMLIGHT with single channel (becuase nothing works with more than one channel)..the 42 LEDS light up, then go red, then heltec pukes. Can't restart it until I pull those lit LEDs...very perplexing. Help? I'm pretty solid with all this stuff. I think its environmental. Admittedly I'm very new to platformio as I've used the arduino ide exclusively for microcontrollers until last week. thanks in advance

Here are my globals for DEMO:

   #define MATRIX_WIDTH            42
    #define MATRIX_HEIGHT           1
    #define NUM_LEDS                (MATRIX_WIDTH*MATRIX_HEIGHT)
    #define NUM_CHANNELS            1
    // #define NUM_RINGS               5
    // #define RING_SIZE_0             24
    #define ENABLE_AUDIO            1
    #define POWER_LIMIT_MW       12 * 10 * 1000   // 10 amp supply at 5 volts assumed

    #define DEFAULT_EFFECT_INTERVAL     (1000*60*1)

    // Once you have a working project, selectively enable various additional features by setting
    // them to 1 in the list below.  This DEMO config assumes no audio (mic), or screen, etc.

    #define ENABLE_WIFI             0   // Connect to WiFi
    #define INCOMING_WIFI_ENABLED   0   // Accepting incoming color data and commands
    #define TIME_BEFORE_LOCAL       0   // How many seconds before the lamp times out and shows local content
    #define ENABLE_NTP              0   // Set the clock from the web
    #define ENABLE_OTA              0   // Accept over the air flash updates

    // #if M5STICKC || M5STICKCPLUS
    //     #define LED_PIN0 32
    // #else
    //     #define LED_PIN0 5
    //     #define LED_PIN1 18
    // #endif

    #define LED_PIN0 5
    // #define LED_PIN1 18
    // #define LED_PIN2 19
    // #define LED_PIN3 17

**Comment by davepl** [comment]
You won’t power 42 LEDs off the regulator on a heltec32.  Use a 5V power brick to power the strip, you should be fine.

**Reply by stretch911** [reply]
if i power the strip separate and apart from the heltec (no shared + or -) the leds go hay wire...see other post. thanks for reply. i'm trying it now, however, it seems like both 5 and 17 (running two 42s off 5v power brick like you said) don't work. if i share the datapin on 5 and run single channel it works.

**Reply by stretch911** [reply]
here's my test. 
single 42 led strip connected to pin5. in global, single channel, pin0 is 5. this works.
keep same physical setup, 42 led to pin5, add in second channel and pin0 defined as 17...compile (w/ adc_deprecated warnings) upload and nothing works. tried it this way powered by the heltec and working and powered by the 5v brick and working on single channel.

each time, when i add in additional channels, led0 pin5 quits working...and i've tried jabbing into 17 and 18, and 19 and 16 (but 16 is shared with oled no?) that's another rathole. 

I really appreciate a reply from THE MAN! I commented on one of your vids I learned basic on trs80 3 and got a trs 80 color computer that Christmas w/ 4k RAM! thanks for all the gifts.

**Comment by davepl** [comment]
Do you have the grounds connected?  You must connected the negative power leg to the negative side of the LED circuit.

**Reply by stretch911** [reply]
i'm drawing a image..you up early boy

**Reply by stretch911** [reply]
![discussion](https://user-images.githubusercontent.com/121883463/210568338-0f54676a-6fd4-4523-931a-de0d3a5c8584.jpg)
test 1 above. (I'm a performance engineer by trade, I'm a QA boy. Thank you for Task Manager!)
![discussion-pin17](https://user-images.githubusercontent.com/121883463/210568464-6fd5e8cf-f4fd-4057-a10e-f3cb86fbf5a1.jpg)
test 2 above to make sure pin 17 actually works.

The only two variables changing in my code is in globals.h/DEMO/NUM_CHANNELS and LED_PIN0 and LED_PIN1

**Reply by stretch911** [reply]
![discussion-pin5a17](https://user-images.githubusercontent.com/121883463/210580971-f9539370-0065-4901-a996-8dcf519f7631.jpg)
just to continue, neither pin works when I enable both. I've even tried this with a single strip, trying both pins 5 and 17. I don't understand. You'd think I'd get a compile error if there was an issue with the channel array.

**Comment by stretch911** [comment]
#if NUM_CHANNELS >= 2
            FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, ((LEDStripGFX *)g_pDevices[0].get())->GetLEDCount());
            pinMode(LED_PIN0, OUTPUT);

            FastLED.addLeds<WS2812B, LED_PIN1, COLOR_ORDER>(g_pDevices[1]->leds, g_pDevices[1]->GetLEDCount());
            pinMode(LED_PIN1, OUTPUT);
        #endif

looks like setup sets up PIN0 and PIN1 as output...
I'm trying to follow the yellow brick road...

**Reply by stretch911** [reply]
#if NUM_CHANNELS == 1
            debugI("Adding %d LEDs to FastLED.", g_pDevices[0]->GetLEDCount());
            
            FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, 3*144);
            FastLED[0].setLeds(((LEDStripGFX *)g_pDevices[0].get())->leds, g_pDevices[0]->GetLEDCount());   
            FastLED.setMaxRefreshRate(0, false);  // turn OFF the refresh rate constraint
            pinMode(LED_PIN0, OUTPUT);
        #endif

this is curious...why is if num_channels ==1 (which works for me) have all this initialization setup contained in the THEN do stuff portion...
when i set my numChannel = 2, it SKIPS this and i just get...

        #if NUM_CHANNELS >= 2
            FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, ((LEDStripGFX *)g_pDevices[0].get())->GetLEDCount());
            pinMode(LED_PIN0, OUTPUT);

            FastLED.addLeds<WS2812B, LED_PIN1, COLOR_ORDER>(g_pDevices[1]->leds, g_pDevices[1]->GetLEDCount());
            pinMode(LED_PIN1, OUTPUT);
        #endif

seems odd...when all the other patterns in this If after If after If seems to build upon one another.... >=2 doesn't do what ==1 does..

**Comment by davepl** [comment]
Try fixing that so that the first (default) one is LED_PIN0 and the second one is LED_PIN1 and then LED_PIN2 and so on.  That’s a bug.

**Reply by stretch911** [reply]
got it...should cascade. thanks

**Reply by stretch911** [reply]
![mainchange](https://user-images.githubusercontent.com/121883463/210788865-ddfa7f35-9f90-488f-9c6c-1c3e6727e60d.jpg)

look right?
   #if USESTRIP

        #if NUM_CHANNELS >= 1
            debugI("Adding %d LEDs to FastLED.", g_pDevices[0]->GetLEDCount());
            
            FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, 3*144);
            FastLED[0].setLeds(((LEDStripGFX *)g_pDevices[0].get())->leds, g_pDevices[0]->GetLEDCount());   
            FastLED.setMaxRefreshRate(0, false);  // turn OFF the refresh rate constraint
            pinMode(LED_PIN0, OUTPUT);
        #endif

        #if NUM_CHANNELS >= 2
            // FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, ((LEDStripGFX *)g_pDevices[0].get())->GetLEDCount());
            // pinMode(LED_PIN0, OUTPUT);

            FastLED.addLeds<WS2812B, LED_PIN1, COLOR_ORDER>(g_pDevices[1]->leds, g_pDevices[1]->GetLEDCount());
            pinMode(LED_PIN1, OUTPUT);
        #endif

**Reply by stretch911** [reply]
nope...not working.... :(

**Comment by stretch911** [comment]
Saga Continued.... After trying about every combination of the above tinkering w/ #if USESTRIP section in main.cpp, I woke this morning and tried this...

    #if USESTRIP

        #if NUM_CHANNELS >= 1
            debugI("Adding %d LEDs to FastLED.", g_pDevices[0]->GetLEDCount());
            debugI("Number of Channels : %d", NUM_CHANNELS);
            
            FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, g_pDevices[0]->GetLEDCount());
            FastLED[0].setLeds(((LEDStripGFX *)g_pDevices[0].get())->leds, g_pDevices[0]->GetLEDCount());   
            FastLED.setMaxRefreshRate(0, false);  // turn OFF the refresh rate constraint
            pinMode(LED_PIN0, OUTPUT);
        #endif

        #if NUM_CHANNELS >= 2
            debugI("Adding %d LEDs to FastLED Channel 2.", g_pDevices[1]->GetLEDCount());
            // // FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, ((LEDStripGFX *)g_pDevices[0].get())->GetLEDCount());
            // // pinMode(LED_PIN0, OUTPUT);

            // FastLED.addLeds<WS2812B, LED_PIN1, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[1].get())->leds, ((LEDStripGFX *)g_pDevices[1].get())->GetLEDCount());
            // FastLED[1].setLeds(((LEDStripGFX *)g_pDevices[1].get())->leds, g_pDevices[1]->GetLEDCount());
            // FastLED.setMaxRefreshRate(0, false);
            //pinMode(LED_PIN1, OUTPUT);
        #endif

So basically, when NUM_CHANNELS =1 in globals.h (DEMO), PIN5 works. Now...When I set NUM_CHANNELS = 2, my debug statement prints the correct info....and EVEN WITH THE CODE COMMENTED, PIN5 and PIN17 DON"T!...VERY ODD>....what happens when NUM_CHANNELS = 2 that makes BOTH pins not work. HELP????!!!!!

**Comment by stretch911** [comment]
Continued....
I added a digitalWrite HIGH and LOW with delay in the channel1 setup and channel2. when NUM_CHANNELS=1, the LED Strip on PIN5 blinks. when NUM_CHANNELS=2 neither pin5 nor pin17 blinks...

        #if NUM_CHANNELS >= 1
            debugI("Adding %d LEDs to FastLED.", g_pDevices[0]->GetLEDCount());
            debugI("Number of Channels : %d", NUM_CHANNELS);
            
            FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, g_pDevices[0]->GetLEDCount());
            FastLED[0].setLeds(((LEDStripGFX *)g_pDevices[0].get())->leds, g_pDevices[0]->GetLEDCount());   
            FastLED.setMaxRefreshRate(0, false);  // turn OFF the refresh rate constraint
            pinMode(LED_PIN0, OUTPUT);
            digitalWrite(LED_PIN0, HIGH);
            delay(1000);
            digitalWrite(LED_PIN0, LOW);
        #endif

        #if NUM_CHANNELS >= 2
            debugI("Adding %d LEDs to FastLED Channel 2.", g_pDevices[1]->GetLEDCount());
            // // FastLED.addLeds<WS2812B, LED_PIN0, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[0].get())->leds, ((LEDStripGFX *)g_pDevices[0].get())->GetLEDCount());
            // // pinMode(LED_PIN0, OUTPUT);

            FastLED.addLeds<WS2812B, LED_PIN1, COLOR_ORDER>(((LEDStripGFX *)g_pDevices[1].get())->leds, ((LEDStripGFX *)g_pDevices[1].get())->GetLEDCount());
            FastLED[1].setLeds(((LEDStripGFX *)g_pDevices[1].get())->leds, g_pDevices[1]->GetLEDCount());
            FastLED.setMaxRefreshRate(0, false);
            pinMode(LED_PIN1, OUTPUT);
            digitalWrite(LED_PIN1, HIGH);
            delay(1000);
            digitalWrite(LED_PIN1, LOW);

        #endif

**Comment by stretch911** [comment]
Problem Solved...
I got to looking back at MAIN.cpp history...and this code hasn't changed since first upload...
scratching my head...i am/was using heltecv2demo

[env:heltecv2demo]                     ; V2 board has 8M flash
board = heltec_wifi_kit_32_v2
monitor_speed = 115200
upload_speed  = 921600
upload_port   = COM9
monitor_port  = COM9
build_flags   = -DDEMO=1
                -DUSE_SCREEN=1
                -DARDUINO_HELTEC_WIFI_KIT_32=1                
                -Dregister=                         ; Sinister:  redefine 'register' so FastLED can use that keyword under C++17
                -std=gnu++17
                -Ofast 

this was my first taste of success...so just stuck with it....got to thinking...what if i don't have a v2? so....tried, ledstrip for the env, but used com9 for upload and build_flags = -DLEDSTRIP
In, effect.cpp for LEDSTRIP is set channels to 2 and pins to 5 and 17....uploaded and now i have 2 independent channels running!!! (with the original main.cpp)

(platformio.ini)
[env:ledstrip]
board         = heltec_wifi_kit_32
build_type    = release
monitor_speed = 115200
upload_speed  = 921600
upload_port   = COM9
monitor_port  = COM9
build_flags   = -DLEDSTRIP=1
                -DUSE_SCREEN=1
                -DARDUINO_HELTEC_WIFI_KIT_32=1
                -std=gnu++17
                -DUNITY_INCLUDE_DOUBLE 
                -DUNITY_DOUBLE_PRECISION=1e-12      ; Make doubles real 8 byte doubles
                -Dregister=                         ; Sinister:  redefined 'register' so FastLED can use that keyword under C++17
                -Os

(globals.h)
#elif LEDSTRIP

    // The LED strips I use for Christmas lights under my eaves

    #define ENABLE_WIFI             1   // Connect to WiFi
    #define INCOMING_WIFI_ENABLED   1   // Accepting incoming color data and commands
    #define WAIT_FOR_WIFI           1   // Hold in setup until we have WiFi - for strips without effects
    #define TIME_BEFORE_LOCAL       5   // How many seconds before the lamp times out and shows local content

    #define NUM_CHANNELS    2
    #define MATRIX_WIDTH    42       // My naximum run, and about all you can do at 30fps  
    #define MATRIX_HEIGHT   1
    #define NUM_LEDS        (MATRIX_WIDTH * MATRIX_HEIGHT)
    #define RESERVE_MEMORY  160000                // WiFi needs about 100K free to be able to (re)connect!
    #define ENABLE_REMOTE   0                     // IR Remote Control
    #define ENABLE_AUDIO    0                     // Listen for audio from the microphone and process it
    #define LED_PIN0        5
    #define LED_PIN1        17

    #define POWER_LIMIT_MW (1000 * 8 * 5)              // Unlimited power for long strips, up to you to limit here or supply enough!

    #define DEFAULT_EFFECT_INTERVAL     (1000*20)    

    #define RING_SIZE_0 1 
    #define RING_SIZE_1 2
    #define RING_SIZE_2 4
    #define RING_SIZE_3 8
    #define RING_SIZE_4 16


whew! just in time. my replacement ws2812 144 2 meters arrive today and i want to get this lamp running!!!!!

Hope this helps someone else out.

**Comment by stretch911** [comment]
added all those effects i'd been working on in DEMO section in effect.cpp (that were working) and then the build blew up. 

c:/users/sever/.platformio/packages/toolchain-xtensa-esp32@8.4.0+2021r2-patch5/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio/build/ledstrip/src/effects.cpp.o:(.literal._ZN13FireFanEffect8DrawFireE10PixelOrder[_ZN13FireFanEffect8DrawFireE10PixelOrder]+0x18): undefined reference to `gVURatio'

.... I then tried env:heltecdemo and that built but nothing is working... :(

.... so i switched env:heltecdemo to use build flag : build_flags = -DLEDSTRIP=1, again, got the gVURatio error. I had pasted all my effects into LEDSTRIP in effect.cpp, so just commented out all of them (that were working...hmm)...back in action w/ single effect. now i have to go thru each effect to find out the culprit....stay tuned :)

#elif LEDSTRIP

        new PaletteEffect(OceanColors_p, 4, 0.1, 0.0, 1.0, 0.0),
        // new TwinkleStarEffect(),
        // new StarryNightEffect<QuietStar>("Ocean Stars", OceanColors_p, STARRYNIGHT_PROBABILITY, 1, LINEARBLEND, 2.0, 0.0, STARRYNIGHT_MUSICFACTOR),
        // new MeteorEffect(1, 1, 5, .15, .25),
        // new StarryNightEffect<BubblyStar>("Bubbly Stars", USAColors_p, STARRYNIGHT_PROBABILITY, 1, LINEARBLEND, 2.0, 0.0, STARRYNIGHT_MUSICFACTOR),
        // new FireFanEffect(NUM_LEDS, 1, 10, 80, 2, 7, Reverse, false, false, true),
        // new ColorFillEffect(CRGB::DarkCyan, 1),
        // new StarryNightEffect<QuietStar>("Rainbow Twinkle Stars", RainbowColors_p, STARRYNIGHT_PROBABILITY, 1, LINEARBLEND, 2.0, 0.0, STARRYNIGHT_MUSICFACTOR),
        // new ColorCycleEffect(Sequential),
        // new StarryNightEffect<BubblyStar>("Little Blooming Rainbow Stars", BlueColors_p, 1.0, 2, LINEARBLEND, 2.0, 0.0, 1), // Blooming Little Rainbow Stars
        // new PaletteEffect(RainbowColors_p, 4, 0.1, 0.0, 1.0, 0.0),
        // new MeteorEffect(6, 1, 25, .15, .12),
        // new FireFanEffect(NUM_LEDS, 1, 10, 80, 2, 7, Reverse, false, false),
        // new MeteorEffect(20, 1, 25, .15, .05),
        // new BouncingBallEffect(3, true, true, 1),
                
        // new RainbowFillEffect(6, 2),

        // new ClassicFireEffect(false, false, 20),

        // new MeteorEffect(12, 1, 25, .15, .08),
        // new RainbowTwinkleEffect(2.0, 14),        
        // new MeteorEffect(),


... enabling each effect one at a time got old pretty quick, so i think that gVURatio error is related to "sound driven effect". Comparing the two Environments in global.h, DEMO had ENABLE_AUDIO set to 1. LEDSTRIP had ENABLE_AUDIO set to 0. Although I'm not using audio (yet), this apparently loads a library that the compiler detects an effect I was using needed....so....skipping the one by one approach, i just set ENABLE_AUDIO to 1, uncommented out all my above effects and BAM!!!! Two channels, rocking these cool effects!!!!!!!!!!!!

i lied....something caused the heltec to reset...curses.

...think it was wifi, i disabled ENABLE_WIFI, INCOMING_WIFI_ENABLED, WAIT_FOR_WIFI, and TIME_BEFORE_LOCAL to zero and i'm on effect number 3 now...both channels (pin5 and 17 working)

Some effects don't work on 2 channels?
new StarryNightEffect<BubblyStar>("Bubbly Stars", USAColors_p, STARRYNIGHT_PROBABILITY, 1, LINEARBLEND, 2.0, 0.0, STARRYNIGHT_MUSICFACTOR),
this guy didn't fire on channel 2....all dark pixels...

new StarryNightEffect<QuietStar>("Rainbow Twinkle Stars", RainbowColors_p, STARRYNIGHT_PROBABILITY, 1, LINEARBLEND, 2.0, 0.0, STARRYNIGHT_MUSICFACTOR),
this guy didn't fire on channel 2...

**Reply by stretch911** [reply]
Final Settings:

platformio.ini

[platformio]
data_dir    = ./data
default_envs = heltecdemo

....

[env:heltecdemo]
board         = heltec_wifi_kit_32
monitor_speed = 115200
upload_speed  = 921600
upload_port   = COM9
monitor_port  = COM9
build_flags   = -DLEDSTRIP=1
                -DUSE_SCREEN=1
                -DARDUINO_HELTEC_WIFI_KIT_32=1                
                -Dregister=                         ; Sinister:  redefine 'register' so FastLED can use that keyword under C++17
                -std=gnu++17
                -Ofast 


globals.h

#elif LEDSTRIP

    // The LED strips I use for Christmas lights under my eaves

    #define ENABLE_WIFI             0   // Connect to WiFi
    #define INCOMING_WIFI_ENABLED   0   // Accepting incoming color data and commands
    #define WAIT_FOR_WIFI           0   // Hold in setup until we have WiFi - for strips without effects
    #define TIME_BEFORE_LOCAL       0   // How many seconds before the lamp times out and shows local content

    #define NUM_CHANNELS    2
    #define MATRIX_WIDTH    42       // My naximum run, and about all you can do at 30fps  
    #define MATRIX_HEIGHT   1
    #define NUM_LEDS        (MATRIX_WIDTH * MATRIX_HEIGHT)
    #define RESERVE_MEMORY  160000                // WiFi needs about 100K free to be able to (re)connect!
    #define ENABLE_REMOTE   0                     // IR Remote Control
    #define ENABLE_AUDIO    1                     // Listen for audio from the microphone and process it
    #define LED_PIN0        5
    #define LED_PIN1        17

    #define POWER_LIMIT_MW (1000 * 8 * 5)              // Unlimited power for long strips, up to you to limit here or supply enough!

    #define DEFAULT_EFFECT_INTERVAL     (1000*30)    

    // #define RING_SIZE_0 1 
    // #define RING_SIZE_1 2
    // #define RING_SIZE_2 4
    // #define RING_SIZE_3 8
    // #define RING_SIZE_4 16

effects.cpp

#elif LEDSTRIP

        new PaletteEffect(OceanColors_p, 4, 0.1, 0.0, 1.0, 0.0),
        new TwinkleStarEffect(),
        //new StarryNightEffect<QuietStar>("Ocean Stars", OceanColors_p, STARRYNIGHT_PROBABILITY, 1, LINEARBLEND, 2.0, 0.0, STARRYNIGHT_MUSICFACTOR),
        new MeteorEffect(1, 1, 5, .15, .25),
        //new StarryNightEffect<BubblyStar>("Bubbly Stars", USAColors_p, STARRYNIGHT_PROBABILITY, 1, LINEARBLEND, 2.0, 0.0, STARRYNIGHT_MUSICFACTOR),
        new FireFanEffect(NUM_LEDS, 1, 10, 80, 2, 7, Reverse, false, false, true),
        new ColorFillEffect(CRGB::DarkCyan, 1),
        //new StarryNightEffect<QuietStar>("Rainbow Twinkle Stars", RainbowColors_p, STARRYNIGHT_PROBABILITY, 1, LINEARBLEND, 2.0, 0.0, STARRYNIGHT_MUSICFACTOR),
        new ColorCycleEffect(Sequential),
        //new StarryNightEffect<BubblyStar>("Little Blooming Rainbow Stars", BlueColors_p, 1.0, 2, LINEARBLEND, 2.0, 0.0, 1), // Blooming Little Rainbow Stars
        new PaletteEffect(RainbowColors_p, 4, 0.1, 0.0, 1.0, 0.0),
        new MeteorEffect(6, 1, 25, .15, .12),
        new FireFanEffect(NUM_LEDS, 1, 10, 80, 2, 7, Reverse, false, false),
        new MeteorEffect(20, 1, 25, .15, .05),
        new BouncingBallEffect(3, true, true, 1),
                
        new RainbowFillEffect(6, 2),

        new ClassicFireEffect(false, false, 20),

        new MeteorEffect(12, 1, 25, .15, .08),
        new RainbowTwinkleEffect(2.0, 14),        
        new MeteorEffect(),

**Comment by davepl** [comment]
I can’t say for sure, but once you’ve given the pin to FastLED I’m not sure you can still manually blink the LED (never tried) but perhaps not.  It really should not matter if no one has called FastLED.show yet

I would try the two code blocks separate.  Change it to 

>     #if NUM_CHANNELS == 1
and
>     #if NUM_CHANNELS == 2

Then try each case to make sure each works separately.  I would use PIN5 and PIN2, not sure if 17 works.  I for sure won’t on a WROVER, I know that.

- Dave

> On Jan 6, 2023, at 4:51 AM, stretc

**Reply by stretch911** [reply]
i got it sorted....i posted each step. it was the board setting :) derrr....i was using heltecv2demo, heltecdemo worked multi channel.
thank you for jumping in. this has been a fantastic learning experience. thanks for the code and the vids.
